docs/CODE_GUIDELINES.md

```diff
@@ -198,80 +217,196 @@ exampleClass      ->InitExample();
 biggerExampleClass->InitExample();
 ```
 
-Right:
+**[back to top](#table-of-contents)**
+
+## 4. Statements
+
+Do not put multiple statements on a single line. Always use a new line for a new statement. It is much easier to debug if one can pinpoint a precise line number.
+
+✅ Good:
+```cpp
+std::vector<std::string> test;
+test.push_back("foobar");
+```
+
+❌ Bad:
 ```cpp
-int value1 = 0;
-int value2 = 0;
-CExampleClass *exampleClass = nullptr;
-CBiggerExampleClass *biggerExampleClass = nullptr;
-exampleClass = new CExampleClass(value1, value2);
-biggerExampleClass = new CBiggerExampleClass(value1, value2);
-exampleClass->InitExample();
-biggerExampleClass->InitExample();
+std::vector<std::string> test; test.push_back("foobar");
 ```
 
 **[back to top](#table-of-contents)**
 
-## 10. Controls Statements
-Insert a new line before every:
-* else in an if statement
-* catch in a try statement
-* while in a do statement
+## 5. Declarations
 
-### 10.1. if else
-Put `then`, `return` or `throw` statements on a new line. Keep `else if` statements on one line.
+### 5.1. Multiple declarations
+Do not put multiple declarations on a single line. This avoids confusion with differing pointers, references, and initialization values on the same line (cf. [ISO C++ guidelines](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es10-declare-one-name-only-per-declaration)).
+
+✅ Good:
 ```cpp
-if (true)
-  return;
-if (true)
-{
-  [...]
-} 
-else if (false)
+char* a;
```

Comment:
I know we've discussed left-aligning or right-aligning, what do other projects/people do and what is their argument?

Reply:
We shouldn't allow that even if right-aligning. It's confusing either way.

Anyway, https://github.com/xbmc/xbmc/pull/14665/files#diff-6c2710b7aab11c31227a420f277f49c7R242

Reply:
I see you included that requirement. Thanks for looking up the existing guidelines, maybe sight these examples and list our reasoning for other projects that are having the same debate?

Reply:
Yeah, because the HP guidelines explicitly state the "what" and "why", I would insert similar to what you said above:

> HP C++ Coding Guidelines say left-align with this reasoning: "The characters * and & are to be written with the type of variables instead of with the name of variables in order to emphasize that they are part of the type definition."